src/flatfile.cpp

```diff
@@ -22,12 +22,12 @@ FlatFileSeq::FlatFileSeq(fs::path dir, const char* prefix, size_t chunk_size) :
 
 std::string FlatFilePos::ToString() const
 {
-    return strprintf("FlatFilePos(nFile=%i, nPos=%i)", nFile, nPos);
+    return strprintf("FlatFilePos(nFile=%i, nPos=%i, nSpan=%i)", nFile, nPos, nSpan);
 }
 
 fs::path FlatFileSeq::FileName(const FlatFilePos& pos) const
 {
-    return m_dir / strprintf("%s%05u.dat", m_prefix, pos.nFile);
+    return m_dir / strprintf("part%02dof%02d/%s%05u.dat", (pos.nFile % pos.nSpan) + 1, pos.nSpan, m_prefix, pos.nFile);
```

Comment:
Here is where the magic happens. Unfortunately, the path is being derived here rather than only the filename which seems to be the original intent of the function. But adding the span prefix to `GetBlocksDir()` seems tricky because the value of nFile isn't available at that time. Any suggestions?